harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -374,9 +374,7 @@ def _auto_step_lr_scheduler_per_batch(
     def _should_update_scaler(self) -> bool:
         if not self.context._scaler or not self.context.experimental._auto_amp:
             return False
-        if self.context.distributed.size > 1:
-            return self.context._should_communicate_and_update()  # type: ignore
-        return True
```

Comment:
So this looks like a breaking change, right?

Existing behavior: non-distributed training would always update the scaler, even if aggregation_frequency was > 1

Proposed behavior: scaler is updated only when the optimizer is stepped, which is based on aggregation_frequency

Is that correct?  It looks like the current behavior is just wrong, but we need to treat it a little specially if it's a breaking change.

Reply:
Ah, now that I have read the thread from the user, I understand this is the key point of the whole PR.

Great!